roqoqo-qryd/src/emulator_devices.rs

```diff
@@ -0,0 +1,622 @@
+    ///
+    /// * `qubit` - The input qubit identifier.
+    ///
+    /// # Returns
+    ///
+    /// * `Ok(usize)` - The tweezer identifier relative to the given qubit.
+    /// * `Err(RoqoqoBackendError)` - If the qubit identifier is not related to any tweezer.
+    pub fn get_tweezer_from_qubit(&self, qubit: &usize) -> Result<usize, RoqoqoBackendError> {
+        self.internal.get_tweezer_from_qubit(qubit)
+    }
+
+    /// Get the names of the available gates in the device.
+    ///
+    /// # Returns
+    ///
+    /// * `Vec<&str>` - Vector of the names of the available gates in the device.
+    /// * `Err(RoqoqoBackendError)` - The given layout name is not present in the device.
+    pub fn get_available_gates_names(&self) -> Result<Vec<&str>, RoqoqoBackendError> {
+        if let Some(available) = self.internal.available_gates.as_ref() {
+            Ok(available.iter().map(|g| g.as_str()).collect())
+        } else {
+            Ok(vec![])
+        }
+    }
+
+    /// Deactivate the given qubit in the device.
+    ///
+    /// # Arguments
+    ///
+    /// * `qubit` - The input qubit identifier.
+    ///
+    /// # Returns
+    ///
+    /// * `Ok(HashMap<usize,usize>)` - The updated qubit -> tweezer mapping.
+    /// * `Err(RoqoqoBackendError)` - If the given qubit identifier is not present in the mapping.
+    pub fn deactivate_qubit(
+        &mut self,
+        qubit: usize,
+    ) -> Result<HashMap<usize, usize>, RoqoqoBackendError> {
+        self.internal.deactivate_qubit(qubit)
+    }
+
+    /// Returns the PhaseShiftedControlledZ phase shift according to the device's relation.
+    ///
+    /// # Returns
+    ///
+    /// * `f64` - The PhaseShiftedControlledZ phase shift.
+    pub fn phase_shift_controlled_z(&self) -> Option<f64> {
+        self.internal.phase_shift_controlled_z()
+    }
+
+    /// Returns the PhaseShiftedControlledPhase phase shift according to the device's relation.
+    ///
+    /// # Returns
+    ///
+    /// * `f64` - The PhaseShiftedControlledPhase phase shift.
+    pub fn phase_shift_controlled_phase(&self, theta: f64) -> Option<f64> {
+        self.internal.phase_shift_controlled_phase(theta)
+    }
+
+    /// Returns the gate time of a PhaseShiftedControlledZ operation with the given qubits and phi angle.
+    ///
+    /// # Arguments
+    ///
+    /// * `control` - The control qubit the gate acts on
+    /// * `target` - The target qubit the gate acts on
+    /// * `phi` - The phi angle to be checked.
+    ///
+    /// # Returns
+    ///
+    /// * `Some<f64>` - The gate time.
+    /// * `None` - The gate is not available on the device.
+    pub fn gate_time_controlled_z(
+        &self,
+        _control: &usize,
+        _target: &usize,
+        phi: f64,
+    ) -> Option<f64> {
+        if let Some(relation_phi) = self.phase_shift_controlled_z() {
+            if (relation_phi.abs() - phi.abs()).abs() < 0.0001 {
+                return Some(1e-6);
+            }
+        }
+        None
+    }
```

Comment:
This doesn't seem to return the gate time of `PhaseShiftedControlledZ` with the given qubits or am I missing something ?

Reply:
thanks, I was still following an old design, now it's fixed